python/ray/dashboard/client/src/pages/dashboard/Tune.tsx

```diff
@@ -0,0 +1,125 @@
+import { Theme } from "@material-ui/core/styles/createMuiTheme";
+import createStyles from "@material-ui/core/styles/createStyles";
+import withStyles, { WithStyles } from "@material-ui/core/styles/withStyles";
+import React from "react";
+import { connect } from "react-redux";
+import { getTuneInfo } from "../../api";
+import { StoreState } from "../../store";
+import { dashboardActions } from "./state";
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+
+const styles = (theme: Theme) =>
+  createStyles({
+    root: {
+      padding: theme.spacing(2),
+      "& > :not(:first-child)": {
+        marginTop: theme.spacing(2)
+      }
+    },
+    table: {
+      marginTop: theme.spacing(1)
+    },
+    cell: {
+      padding: theme.spacing(1),
+      textAlign: "center",
+      "&:last-child": {
+        paddingRight: theme.spacing(1)
+      }
+    }
+  });
+
+const mapStateToProps = (state: StoreState) => ({
+  tuneInfo: state.dashboard.tuneInfo,
+});
+
+const mapDispatchToProps = dashboardActions;
+
+class Tune extends React.Component<WithStyles<typeof styles> &
+  ReturnType<typeof mapStateToProps> &
+  typeof mapDispatchToProps> {
+  interval: number = 0;
+
+  refreshTuneInfo = async () => {
+    try {
+      const [tuneInfo] = await Promise.all([
+        getTuneInfo()
+      ]);
+      this.props.setTuneInfo({ tuneInfo });
+    } catch (error) {
+      this.props.setError(error.toString());
+    }
+  };
+
+  async componentDidMount() {
+    this.interval = window.setInterval(this.refreshTuneInfo, 1000);
+  }
+
+  async componentWillUnmount() {
+    window.clearInterval(this.interval);
+  }
+
+  render() {
+    const { classes, tuneInfo } = this.props;
+
+    if (tuneInfo === null || Object.keys(tuneInfo["trial_records"]).length === 0) {
+      return null;
+    }
+
+
+    let paramNames: string[] = [];
+    if (tuneInfo != null) {
+      let paramsDict = tuneInfo["trial_records"][Object.keys(tuneInfo["trial_records"])[0]]["params"];
```

Comment:
since `Object.keys(tuneInfo["trial_records"])[0]` is shared between `paramsDict` and `metricNames` construction, let's move it to a separate variable for clarity